src/client.rs

```diff
@@ -111,23 +116,15 @@ impl SGClient {
     /// }
     /// ```
     #[cfg(not(feature = "async"))]
-    pub fn send(&self, mail_info: Mail) -> SendgridResult<String> {
+    pub fn send(&self, mail_info: Mail) -> SendgridResult<Response> {
         let post_body = make_post_body(mail_info)?;
-        let res = self
+        Ok(self
             .client
             .post(API_URL)
             .headers(self.headers()?)
             .body(post_body)
-            .send()?;
-
-        if !res.status().is_success() {
-            Err(SendgridError::RequestNotSuccessful(
-                res.status(),
-                res.text()?,
-            ))
-        } else {
-            Ok(res.text()?)
-        }
+            .send()
+            .and_then(|x| x.error_for_status())?)
```

Comment:
When getting an "invalid status", body of the response would be something like:
```json
{"errors":[{"message":"The content type must be a string at least one character in length.","field":"content.0.type","help":"http://sendgrid.com/docs/API_Reference/Web_API_v3/Mail/errors.html#message.content.type"}]}
```
> @gsquire please check the latest commit, it should fix: #66 #65

On the contrary, e4700e8b8640917cdf297c5c5bc23c5dd48de827 removes the ability for the user to extract the body from the error. I think what was done initially with
```rust
#[error("Request failed with StatusCode: `{0:?}, Body: `{1:?}")]
RequestNotSuccessful(reqwest::StatusCode, String),
```
was perfectly good for debugging, though I'm not sure why the `Display` formatters were replaced by `Debug` formatters, and would probably have favored named fields in the enum to make the meaning of the `String` more obvious:
```rust
pub enum SendgridError {
    ...
    RequestNotSuccessful {
        status_code: reqwest::StatusCode,
        body: String,
    },
}
```

Reply:
so you want body

Reply:
yes

Reply:
> I created an issue at: seanmonstar/reqwest#1043, I think it's better to make it to be addressed in reqwest.
Wasn't this supposed to be in this conversation?